src/services/UserValidationService.ts

```diff
@@ -1,42 +1,57 @@
 import { Service } from '@tsed/di';
 import { Unauthorized } from '@tsed/exceptions';
-import * as bcrypt from 'bcrypt';
 
-import { User } from '../entities/User';
+import { UserCreation } from '../models/UserCreation';
+import { UserLogin } from '../models/UserLogin';
 
 @Service()
 export class UserValidationService {
 
-	public async validateUser(
-		userToValidate: User, 
-	): Promise<User> {
+	public validateUserLogin(userLogin: UserLogin): void {
+		const email: string= userLogin.email;
+		const password: string = userLogin.password;
+		
+		this.checkParameters([email, password]);
+		this.checkEmail(email);
+		this.checkPassword(password);
+	}
+
+	public validateUserCreation(userCreation: UserCreation): void {
 
-		const username: string = userToValidate.username;
-		const email: string= userToValidate.email;
-		const password: string = userToValidate.password;
+		const username: string = userCreation.username;
+		const email: string= userCreation.email;
+		const password: string = userCreation.password;
 		
-		if (Object.keys(userToValidate).length === 0) {
-			throw new Unauthorized("Empty request body!");
-		}
-	
-		if (!username || !email || !password) {
-			throw new Unauthorized("Missing credentials!");
-		}
+		this.checkParameters([username, email, password]);
```

Comment:
Similarly to previous comment, checking here is used multiple times and it is not really clear what is checked. At first, `parameters` are checked, but later all parameters are `checked` separately. I believe there can be a better naming.

Whenever you are writing code, it may be good to put yourself in a role of someone in 2 months who will be going through this. What would you know from these names without any additional context?

Reply:
I dont understand here. I was following the ticket on Jira. "What if request body is empty?" From my point of view I am first checking if request have even some body (with any parameters) so if not and request is badly sended and its empty it will terminate checking here so it will not continue further. (Performence?) Idk. And after that I am checking if the RB have even the right properties in it.